src/Application/Query.php

```diff
@@ -29,4 +34,12 @@ public function getFreeText(): string {
 		return $this->freeText;
 	}
 
+	public function getInstancePropertyId(): ?PropertyId {
+		return $this->instance['propertyId'] ?? null;
```

Comment:
This information is at Config->instanceOfId, we do not need to parse it out of the string

Reply:
Did not know that, will update it tomorrow